src/app/accounts/set-default-walletid.ts

```diff
@@ -0,0 +1,22 @@
+import { InvalidWalletId } from "@domain/errors"
+import { AccountsRepository } from "@services/mongoose"
+
+export const setDefaultWalletId = async ({
+  accountId,
+  walletId,
+}: {
+  accountId: AccountId
+  walletId: WalletId
+}) => {
+  const account = await AccountsRepository().findById(accountId)
+  if (account instanceof Error) return account
+
+  if (!account.walletIds.some((wid) => wid === walletId)) return InvalidWalletId
```

Comment:
should this be `return new InvalidWalletId()`? (another reason to add return type to the method)